astropy/coordinates/tests/test_representation.py

```diff
@@ -1223,6 +1223,16 @@ def test_transform(self):
         assert ds2.d_y.unit == u.km / u.s
         assert ds2.d_z.unit == u.km / u.s
 
+    def test_transform_non_contiguous_matrix(self):
+        # Regression test for gh-15503 (due to pyerfa gh-123)
+        s1 = CartesianRepresentation([1, 2, 3] * u.kpc)
+        matrix = np.array([[1, 0, 0, 5], [0, 1, 0, 6], [0, 0, 1, 7]], dtype="float")[
+            :, :3
+        ]
+        assert_array_equal(matrix, np.eye(3))
+        s2 = s1.transform(matrix)
+        assert representation_equal(s2, s1)
```

Comment:
This doesn't have to take up so many lines:
```suggestion
        vector = CartesianRepresentation([1, 2, 3] * u.kpc)
        view = np.array([[1, 0, 0, 5], [0, 1, 0, 6], [0, 0, 1, 7]], dtype=float)[:, :3]
        assert_array_equal(view, np.eye(3))
        assert representation_equal(vector.transform(view), vector)
```

Reply:
Indeed a bit better, so I implemented it (though with even shorter names). And I guess worth the cycles since a rebase was needed anyway....